adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -123,6 +125,34 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] extends Logging {
   val sequences: SequenceDictionary
 
   /**
+   * Replaces the sequence dictionary attached to a GenomicRDD.
+   *
+   * @param newSequences The new sequence dictionary to attach.
+   * @return Returns a new GenomicRDD with the sequences replaced.
+   */
+  def replaceSequences(newSequences: SequenceDictionary): U
+
+  /**
+   * Appends sequence metadata to the current RDD.
+   *
+   * @param sequencesToAdd The new sequences to append.
+   * @return Returns a new GenomicRDD with the sequences appended.
+   */
+  def addSequences(sequencesToAdd: SequenceDictionary): U = {
```

Comment:
I mean, `SequenceDictionary` supports the `++` operator, so my preference is not to make the function variadic. Mostly, it adds a bit of error checking that we need to do (WRT being provided no sequencesToAdd).